### ISSUE TITLE
Support for unordered sequence equality

### ISSUE THREAD

**Issue by megakid** [diegofrata/Generator.Equals#1]

**Comment by megakid** [comment]
```c#
using System;
using System.Collections.Generic;
using System.Linq;

namespace Bleb
{
    public class UnorderedEnumerableEqualityComparer<T> : IEqualityComparer<IEnumerable<T>>
    {
        private readonly IEqualityComparer<T> _equalityComparer;

        // ReSharper disable once InconsistentNaming
        public static IEqualityComparer<IEnumerable<T>> Default { get; } = new UnorderedEnumerableEqualityComparer<T>();

        public UnorderedEnumerableEqualityComparer()
            : this(EqualityComparer<T>.Default)
        {
        }

        public UnorderedEnumerableEqualityComparer(IEqualityComparer<T> equalityComparer)
        {
            _equalityComparer = equalityComparer;
        }

        public bool Equals(IEnumerable<T> x, IEnumerable<T> y)
        {
            if (x == null && y == null) return true;
            if (x == null || y == null) return false;
            // Orderless.

            var cnt = new Dictionary<T, int>(_equalityComparer);

            foreach (var s in x)
            {
                if (cnt.ContainsKey(s))
                {
                    cnt[s]++;
                }
                else
                {
                    cnt.Add(s, 1);
                }
            }

            foreach (var s in y)
            {
                if (cnt.ContainsKey(s))
                {
                    cnt[s]--;
                }
                else
                {
                    return false;
                }
            }

            return cnt.Values.All(c => c == 0);
        }

        public int GetHashCode(IEnumerable<T> obj)
        {
            if (obj == null) return 0;

            // commutative (orderless)
            return obj.Aggregate(19,
                (current, o) =>
                {
                    var hash = o == null ? 0 : _equalityComparer.GetHashCode(o);
                    hash = hash == 0 ? 1 : hash;

                    return unchecked((current * 397) * hash);
                });
        }
    }
}
```

**Comment by diegofrata** [comment]
Released in 0.4.0.
